mentoring/answer.py

```diff
@@ -70,43 +174,39 @@ def student_input(self):
 
         return student_input
 
-    def mentoring_view(self, context=None):
-        if not self.read_only:
-            html = render_template('templates/html/answer_editable.html', {
-                'self': self,
-            })
-        else:
-            html = render_template('templates/html/answer_read_only.html', {
-                'self': self,
-            })
-        
-        fragment = Fragment(html)
-        fragment.add_css_url(self.runtime.local_resource_url(self.xblock_container, 'public/css/answer.css'))
-        fragment.add_javascript_url(self.runtime.local_resource_url(self.xblock_container,
-                                                                    'public/js/answer.js'))
-        fragment.initialize_js('AnswerBlock')
-        return fragment
+    def fallback_view(self, view_name, context=None):
+        context = context or {}
+        context['self'] = self
+        html = loader.render_template('templates/html/answer_editable.html', context)
 
-    def mentoring_table_view(self, context=None):
-        html = render_template('templates/html/answer_table.html', {
-            'self': self,
-        })
         fragment = Fragment(html)
-        fragment.add_css_url(self.runtime.local_resource_url(self.xblock_container, 'public/css/answer_table.css'))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/answer.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/answer.js'))
+        fragment.initialize_js('AnswerBlock')
         return fragment
 
     def submit(self, submission):
-        if not self.read_only:
-            self.student_input = submission[0]['value'].strip()
-            log.info(u'Answer submitted for`{}`: "{}"'.format(self.name, self.student_input))
+        self.student_input = submission[0]['value'].strip()
+        self.save()
```

Comment:
If you aren't handling a save failure, why call `save` aggressively here, rather than just letting the runtime do it for you once the handler completes?